scripts/consensus_merge.py

```diff
@@ -0,0 +1,636 @@
+            return '1/1'
+        elif name == 'het':
+            return '0/1'
+        elif name in ('ref', 'conflict'):
+            return '0/0'
+        else:
+            raise ValueError('No genotype associated with %s' % name)
+
+    def __init__(self, name, stype, record, caller):
+        """ Initialize this object and calculate 
+            Args:
+                name (str): the name of the sample from the VCF
+                stype (str): sample may be 'tumor' or 'normal'
+                record (pysam.VariantRecord): record with which this
+                sample is associated
+                caller (str): name of caller for this variant, 
+                    must be a member of CALLER_NAMES (case insensitive)
+
+            Raises:
+                ValueError if stype is not 'tumor' or 'normal'
+                IOError if name is not associated with exactly one
+                    sample in record
+                ValueError if caller is not in CALLER_NAMES
+        """
+        self.name = name
+        if stype not in ('tumor', 'normal'):
+            raise ValueError('Sample type must be tumor or normal, not %s' % stype)
+        self.type = stype
+        self.record = record
+
+        target_sample = [self.record.samples[index] for index in range(len(self.record.samples)) 
+                         if self.record.samples[index].name == self.name]
+        if len(target_sample) != 1:
+            raise IOError('Sample name %s not found or duplicated' % self.name)
+        self.vcf_sample = target_sample[0]
+
+        caller_test = [c.lower() for c in CALLER_NAMES]
+        if caller.lower() not in caller_test:
+            raise ValueError('Unknown caller %s' % caller)
+
+        self.caller = caller.lower()
+
+    @functools.cached_property
+    def GT(self):
+        """ Instructions for GT FORMAT attribute """
+        if self.caller == 'strelka2':
+            if self.type == 'normal':
+                strelka_nt = self.record.info['NT']
+                return Sample.name_to_gt(strelka_nt)
+            elif self.type == 'tumor':
+                # Strelka2 SGT field may be e.g. 'ref->het' or 'GG->GC'
+                strelka_sgt = self.record.info['SGT'].split("->")[-1]
+                try:
+                    tumor_gt = Sample.name_to_gt(strelka_sgt)
+                except ValueError:
+                    if len(set(strelka_sgt)) == 2:
+                        tumor_gt = '0/1'
+                    elif len(set(strelka_sgt)) == 1:
+                        if strelka_sgt[0] == self.record.ref:
+                            tumor_gt = '0/0'
+                        else:
+                            tumor_gt = '1/1'
+                    else:
+                        raise IOError('Unhandled Strelka2 SGT value %s' % record.info['SGT'])
+                return tumor_gt
+
+        elif self.caller in ('mutect2', 'vardict', 'lancet'):
+            return stringify(sorted(self.vcf_sample['GT']), '/')
```

Comment:
It stores it in the order in which it appears, but at least one caller, Mutect2 I think, has 1/0 genotypes in it, which are then treated as distinct from 0/1 unless sorted.

Reply:
Yeah, and VarDict does it too - they try to use it as an "informative" heterozygous GT in that, if AF is below a cutoff, its `0/1`, otherwise it's `1/0`